forem/forem
# Use custom, dynamic, error pages

## What type of PR is this? (check all applicable)


## Description

This PR moves the static 404, 422, 500 and 503 pages that we have to be handled dynamically.

It closes https://github.com/forem/forem/issues/11718 which was requested by @pkfrank to avoid showing the prompt to open an issue on Forems that aren't DEV.

It also opens the path to customization aka crayonsifications of the error pages.

Kudos to @justalever for the great tutorial on DEV: https://dev.to/justalever/custom-error-pages-in-ruby-on-rails-5270

## Related Tickets & Documents

#11718 

## QA Instructions, Screenshots, Recordings

1. set `config.consider_all_requests_local` to `false` in the file `config/environments/development.rb` file
1. load the web server
1. navigate to either http://localhost:3000/404, http://localhost:3000/422, http://localhost:3000/500, http://localhost:3000/503 to verify they work and are indeed the old pages, same as https://dev.to/404, https://dev.to/422, https://dev.to/500 and https://dev.to/503

To test wether the prompt to open an issue appears or not, you either need to change your app domain to `dev.to` or `forem.dev` or add `localhost:3000` to `ErrorsController::GITHUB_BUG_REPORT_DOMAINS` and then visit http://localhost:3000/500

### UI accessibility concerns?

nope, though we might want to check those pages for a11y in a separate PR

## Added tests?

      have not been included_
